src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/MutationScripts/Join.txt

```diff
@@ -0,0 +1,55 @@
+﻿#SETUP: PowerFxV1CompatibilityRules,StronglyTypedBuiltinEnums, SupportColumnNamesAsIdentifiers
+
+>> Set(t1, 
+  Table({Id:"AAA111",Name:"Contoso",Country:"USA"},
+  {Id:"BBB222",Name:"Fabrikam",Country:"CAN"},
+  {Id:"EEE555",Name:"ACME INC.",Country:"MEX"},
+  {Id:"FFF666",Name:"Dunder Mifflin",Country:"USA"}))
+Table({Country:"USA",Id:"AAA111",Name:"Contoso"},{Country:"CAN",Id:"BBB222",Name:"Fabrikam"},{Country:"MEX",Id:"EEE555",Name:"ACME INC."},{Country:"USA",Id:"FFF666",Name:"Dunder Mifflin"})
+
+>> Set(t2, 
+  Table({SupplierId:"AAA111", Fruit:"Grapes", Price:220, Purchase:Date(2015,10,1), RowId:1},
+  {SupplierId:"BBB222", Fruit:"Lemons", Price:31, Purchase:Date(2015,10,1), RowId:2},
+  {SupplierId:"AAA111", Fruit:"Lemons", Price:29, Purchase:Date(2015,10,2), RowId:3},
+  {SupplierId:"AAA111", Fruit:"Grapes", Price:210, Purchase:Date(2015,10,2), RowId:4},
+  {SupplierId:"BBB222", Fruit:"Lemons", Price:30, Purchase:Date(2015,10,3), RowId:5},
+  {SupplierId:"AAA111", Fruit:"Bananas", Price:12, Purchase:Date(2015,10,3), RowId:6},
+  {SupplierId:"CCC333", Fruit:"Apples", Price:15, Purchase:Date(2015,12,3), RowId:7},
+  {SupplierId:"DDD444", Fruit:"Mangos", Price:30, Purchase:Date(2016,12,3), RowId:8}))
+Table({Fruit:"Grapes",Price:220,Purchase:Date(2015,10,1),RowId:1,SupplierId:"AAA111"},{Fruit:"Lemons",Price:31,Purchase:Date(2015,10,1),RowId:2,SupplierId:"BBB222"},{Fruit:"Lemons",Price:29,Purchase:Date(2015,10,2),RowId:3,SupplierId:"AAA111"},{Fruit:"Grapes",Price:210,Purchase:Date(2015,10,2),RowId:4,SupplierId:"AAA111"},{Fruit:"Lemons",Price:30,Purchase:Date(2015,10,3),RowId:5,SupplierId:"BBB222"},{Fruit:"Bananas",Price:12,Purchase:Date(2015,10,3),RowId:6,SupplierId:"AAA111"},{Fruit:"Apples",Price:15,Purchase:Date(2015,12,3),RowId:7,SupplierId:"CCC333"},{Fruit:"Mangos",Price:30,Purchase:Date(2016,12,3),RowId:8,SupplierId:"DDD444"})
+
+// Inner
+>> Join(t1, t2, LeftRecord.Id = RightRecord.SupplierId, JoinType.Inner)
```

Comment:
>t1, t2 [](http://example.com/codeflow?start=8&length=6)

We should have tests for renaming the left/right scopes:

    Join(
        t1 As MyLeft,
        t2 As MyRight,
        MyLeft.Id = MyRight.SupplierId,
        JoinType.Inner,
        MyRight.Fruit As Fruit)